src/test/java/com/ebivariation/contigalias/controller/ContigAliasControllerIntegrationTest.java

```diff
@@ -56,56 +57,107 @@
     @MockBean
     private ChromosomeService mockChromosomeService;
 
-    @BeforeEach
-    void setUp() {
-        when(mockAssemblyService.getAssemblyByAccession(entity.getGenbank()))
-                .thenReturn(Optional.of(entity));
-        when(mockAssemblyService.getAssemblyByGenbank(entity.getGenbank()))
-                .thenReturn(Optional.of(entity));
-        when(mockAssemblyService.getAssemblyByRefseq(entity.getRefseq()))
-                .thenReturn(Optional.of(entity));
+    @Nested
+    class AssemblyServiceTests {
+
+        private final AssemblyEntity entity = AssemblyGenerator.generate();
+
+
+        @BeforeEach
+        void setUp() {
+            when(mockAssemblyService.getAssemblyByAccession(entity.getGenbank()))
+                    .thenReturn(Optional.of(entity));
+            when(mockAssemblyService.getAssemblyByGenbank(entity.getGenbank()))
+                    .thenReturn(Optional.of(entity));
+            when(mockAssemblyService.getAssemblyByRefseq(entity.getRefseq()))
+                    .thenReturn(Optional.of(entity));
+        }
+
+        @Test
+        void getAssemblyByAccessionGCAHavingChromosomes() throws Exception {
+            ResultActions resultActions = mockMvc.perform(
+                    get("/contig-alias/assemblies/{accession}", entity.getGenbank()));
+            testAssemblyIdenticalToEntity(resultActions);
+        }
+
+        @Test
+        void getAssemblyByGenbank() throws Exception {
+            ResultActions resultActions = mockMvc.perform(
+                    get("/contig-alias/assemblies/genbank/{genbank}", entity.getGenbank()));
+            testAssemblyIdenticalToEntity(resultActions);
+        }
+
+        @Test
+        void getAssemblyByRefseq() throws Exception {
+            ResultActions resultActions = mockMvc.perform(
+                    get("/contig-alias/assemblies/refseq/{refseq}", entity.getRefseq()));
+            testAssemblyIdenticalToEntity(resultActions);
+        }
```

Comment:
this tests are fine as they are now, but for future code, remember you don't need to do the same tests twice (once in the controller integration test, and again in the controller unit tests).

The integration tests are meant to test that everything is wired properly and check things at the ends of the processes, like the serialization, the autowiring, etc. Most of the other things can be tested only in the unit tests.